src/components/standalone/openvpn_tunnel/CreateOrEditTunnelDrawer.vue

```diff
@@ -0,0 +1,846 @@
+const presharedKey = ref('')
+const protocol = ref<'tcp' | 'udp'>('udp')
+const compression = ref('disabled')
+const digest = ref('auto')
+const cipher = ref('auto')
+
+// Server tunnel form fields
+const publicEndpoints = ref<string[]>([''])
+const localNetworks = ref<NeComboboxOption[]>([])
+const minimumTLSVersion = ref('auto')
+const vpnNetwork = ref('')
+
+// Client tunnel form fields
+const remoteHosts = ref<string[]>([''])
+const authentication = ref<'certificate' | 'username_password_certificate'>('certificate')
+const username = ref('')
+const password = ref('')
+const certificate = ref('')
+const mode = ref<'bridged' | 'routed'>('bridged')
+
+// remote options
+const digestOptions = ref<NeComboboxOption[]>([])
+const cipherOptions = ref<NeComboboxOption[]>([])
+const localNetworksOptions = ref<NeComboboxOption[]>([])
+
+const compressionOptions = [
+  {
+    id: 'disabled',
+    label: t('standalone.openvpn_tunnel.disabled')
+  },
+  {
+    id: 'lz0',
+    label: 'LZ0'
+  },
+  {
+    id: 'lz4',
+    label: 'LZ4'
+  }
+]
+
+const topologyOptions = [
+  {
+    id: 'subnet',
+    label: t('standalone.openvpn_tunnel.subnet')
+  },
+  {
+    id: 'p2p',
+    label: t('standalone.openvpn_tunnel.p2p')
+  }
+]
+
+const authenticationOptions = [
+  {
+    id: 'certificate',
+    label: t('standalone.openvpn_tunnel.certificate')
+  },
+  {
+    id: 'username_password_certificate',
+    label: t('standalone.openvpn_tunnel.username_password_certificate')
+  }
+]
+
+const modeOptions = [
+  {
+    id: 'bridged',
+    label: t('standalone.openvpn_tunnel.bridged')
+  },
+  {
+    id: 'routed',
+    label: t('standalone.openvpn_tunnel.routed')
+  }
+]
+
+const protocolOptions = [
+  {
+    id: 'udp',
+    label: t('standalone.openvpn_tunnel.udp')
+  },
+  {
+    id: 'tcp',
+    label: t('standalone.openvpn_tunnel.tcp')
+  }
+]
+
+const tlsOptions = [
+  {
+    id: 'auto',
+    label: t('standalone.openvpn_tunnel.auto')
+  },
+  {
+    id: '1.1',
+    label: '1.1'
+  },
+  {
+    id: '1.2',
+    label: '1.2'
+  }
+]
+
+async function fetchOptions() {
+  try {
+    loading.value = true
+    cipherOptions.value = [
+      {
+        id: 'auto',
+        label: t('standalone.openvpn_tunnel.auto'),
+        description: `(${t('standalone.openvpn_tunnel.server_client_negotiation')})`
+      },
+      ...(await ubusCall('ns.ovpntunnel', 'list-cipher')).data.ciphers.map(
+        (cipher: { name: string; description: string }) => ({
+          id: cipher.name,
+          label: cipher.name
+        })
+      )
+    ]
+    digestOptions.value = [
+      {
+        id: 'auto',
+        label: t('standalone.openvpn_tunnel.auto'),
+        description: `(${t('standalone.openvpn_tunnel.server_client_negotiation')})`
+      },
+      ...(await ubusCall('ns.ovpntunnel', 'list-digest')).data.digests.map(
+        (digest: { name: string; description: string }) => ({
+          id: digest.name,
+          label: digest.name
+        })
+      )
+    ]
```

Comment:
Good as one-liners, not so good at readability, I think a separation of these or additional comments are needed

Reply:
Tried adding some comments to better describe the lists' components. Let me know if this can work (we could also simply separate the components, but in this case I think a comment should be enough)

Reply:
It should be fine, I was suggesting it just so that we don't have to ping pong back and fourth from the back-end doc to the UI